src/opam.ml

```diff
@@ -275,6 +275,11 @@ let upgrade t switch = spawn t [ "upgrade"; switch_arg switch; "-y" ]
 let remove t switch packages =
   spawn t ("remove" :: switch_arg switch :: "-y" :: packages)
 
+let path (opam : t) (switch : Switch.t) =
+  match switch with
+  | Local p -> Path.(p / "_opam")
+  | Named n -> Path.(opam.root / n)
```

Comment:
That doesn't seem right to me. Shouldn't we be running `opam var prefix` or something?

Reply:
That function was copied over from the `Opam.Switch` module. I was struggling to expose it in the `.mli` file so I decided to copy it. I've fixed that now by moving the `type t` at the top of the file and by using a "type substitution" `type opam := t`. See https://github.com/ocamllabs/vscode-ocaml-platform/pull/771/commits/76b52651dc805ecc2a70ca006d6d09badc5475be

We can revert that change and not expose the `path` function if we decide not to provide a `default_cache_dir`. See my comment here: https://github.com/ocamllabs/vscode-ocaml-platform/pull/771#discussion_r760440462

Reply:
Removed that code